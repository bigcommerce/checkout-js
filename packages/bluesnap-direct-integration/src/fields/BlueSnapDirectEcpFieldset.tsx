import { LanguageService } from '@bigcommerce/checkout-sdk';
import React, { FunctionComponent, memo } from 'react';

import BlueSnapDirectNumberField from './BlueSnapDirectNumberField';
import BlueSnapDirectSelectField from './BlueSnapDirectSelectField';

export interface BlueSnapDirectEcpFieldsetProps {
    language: LanguageService;
    useFloatingLabel?: boolean;
}

const BlueSnapDirectEcpFieldset: FunctionComponent<BlueSnapDirectEcpFieldsetProps> = ({
    language,
    useFloatingLabel,
}) => {
    const options = {
        helperLabel: language.translate('payment.bluesnap_direct_account_type_select.label'),
        items: [
            {
                label: language.translate(
                    'payment.bluesnap_direct_account_type_select.option_consumer_checking',
                ),
                value: 'CONSUMER_CHECKING',
            },
            {
                label: language.translate(
                    'payment.bluesnap_direct_account_type_select.option_consumer_savings',
                ),
                value: 'CONSUMER_SAVINGS',
            },
            {
                label: language.translate(
                    'payment.bluesnap_direct_account_type_select.option_corporate_checking',
                ),
                value: 'CORPORATE_CHECKING',
            },
            {
                label: language.translate(
                    'payment.bluesnap_direct_account_type_select.option_corporate_savings',
                ),
                value: 'CORPORATE_SAVINGS',
            },
        ],
    };

    return (
        <>
            <BlueSnapDirectNumberField
                labelContent={language.translate('payment.bluesnap_direct_account_number.label')}
                maxLength={17}
                name="accountNumber"
                useFloatingLabel={useFloatingLabel}
            />

            <BlueSnapDirectNumberField
                labelContent={language.translate('payment.bluesnap_direct_routing_number.label')}
                maxLength={9}
                name="routingNumber"
                useFloatingLabel={useFloatingLabel}
            />

            <BlueSnapDirectSelectField
                labelContent={language.translate('payment.bluesnap_direct_account_type.label')}
                name="accountType"
                options={options}
                useFloatingLabel={useFloatingLabel}
            />
        </>
    );
};

export default memo(BlueSnapDirectEcpFieldset);
