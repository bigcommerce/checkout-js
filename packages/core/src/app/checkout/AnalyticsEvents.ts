let boltTracker: Window["BoltTrack"] = {
  recordEvent: (event: string) => console.log("--bolt bigc--: NOOP ", event),
};

const eventLog: string[] = [];

export const AnalyticsEvents = {
  init: () => {
    try {
      console.log("--bolt bigc--: intializing Bolt-BigC analytics events");
      if (window && window.BoltTrack) {
        boltTracker = window.BoltTrack;
        console.log("--bolt bigc--: successfully assigned window BoltTrack");
        // immediately send checkout load success event
        AnalyticsEvents.emitEvent("Checkout load success");
      }
    } catch (e) {
      console.log("--bolt bigc--: error during emitting event ", e);
    }
  },
  emitEvent: (eventName: string) => {
    try {
      console.log("--bolt bigc--: emitting analytics event ", eventName);
      const props: any = {
        nextState: eventName,
        prevState: "",
      };
      if (eventLog.length > 0) {
        props.prevState = eventLog[0];
      }
      boltTracker.recordEvent("CheckoutFunnelTransition", props);
      // add latest event to beginning of log array
      eventLog.unshift(eventName);
    } catch (e) {
      console.log("--bolt bigc--: error during emitting event ", e);
    }
  },
  onBeforeUnload: () => {
    AnalyticsEvents.emitEvent("Exit");
  },
};
