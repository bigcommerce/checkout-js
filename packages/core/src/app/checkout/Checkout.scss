@import '../ui/Base';

$color-secondary: #757575;
$color-link: #476BEF;
$fontFamily-montserrat: 'Montserrat';

// optimizedCheckout-headingPrimary
.header, .header-secondary, .sub-header {
    color: $color-black;
    font-family: $fontFamily-montserrat;
    font-size: fontSize("large");
    font-weight: fontWeight("bold");
}

.header-secondary {
    color: $color-secondary;
}

// optimizedCheckout-headingSecondary
.sub-header {
    font-size: fontSize("smaller");
}

// optimizedCheckout-contentPrimary
.body-regular, .body-cta, .body-medium, .body-bold {
    color: $color-black;
    font-family: $fontFamily-montserrat;
    font-size: fontSize("base");
    font-weight: fontWeight("normal");
}

.body-cta {
    color: $color-link;
}

.body-medium, .body-bold {
    font-weight: fontWeight("medium");
}

.body-bold {
    font-size: fontSize("base");
    font-weight: fontWeight("bold");
}

.sub-text, .sub-text-medium, .sub-text-bold {
    color: $color-black;
    font-family: $fontFamily-montserrat;
    font-size: fontSize("smallest");
    font-weight: fontWeight("normal");
}

.sub-text-medium {
    color: $color-secondary;
    font-weight: fontWeight("medium");
}

.sub-text-bold {
    font-weight: fontWeight("bold");
}

// overrides 
.body-bold, .body-medium {
    &.button--primary:not(:disabled){
        color: $color-white;
    }
}

.form-checkbox + .form-label,
.form-radio + .form-label {
    &.body-regular {
        color: $color-black;
        font-family: $fontFamily-montserrat;
        font-size: fontSize("base");
        font-weight: fontWeight("normal");
    }
}

.shippingOptionLabel {
    .shippingOption-price, .shippingOption-desc {
        &.body-medium {
            color: $color-black;
            font-family: $fontFamily-montserrat;
            font-size: fontSize("base");
            font-weight: fontWeight("medium");
        }
    }
}

.body-medium {
     small {
        color: $color-black;
        font-family: $fontFamily-montserrat;
        font-size: fontSize("base");
        font-weight: fontWeight("medium");
    }
}

.add-consignment-button {
    &.body-regular {
        color: $color-black;
    }
}
