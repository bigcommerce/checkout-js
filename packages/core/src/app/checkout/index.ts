export { RenderCheckout, RenderCheckoutOptions } from './renderCheckout';
export { default as CheckoutContext, CheckoutContextProps } from './CheckoutContext';
export {
    default as CheckoutProvider,
    CheckoutProviderProps,
    CheckoutProviderState,
} from './CheckoutProvider';
export { default as withCheckout, WithCheckoutProps } from './withCheckout';
export { default as CheckoutSupport } from './CheckoutSupport';
export { default as NoopCheckoutSupport } from './NoopCheckoutSupport';
export { default as navigateToOrderConfirmation } from './navigateToOrderConfirmation';
