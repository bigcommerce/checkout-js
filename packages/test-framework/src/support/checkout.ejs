<!DOCTYPE html>
<html>
    <head>
        <title>Local Testing Environment</title>
        <base href="/" target="_top"/>
        <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1" />
        <link href="https://fonts.googleapis.com/css?family=Karla:400|Montserrat:400,500,700&display=swap" rel="stylesheet" />
    </head>
    <body>
    <header class="checkoutHeader optimizedCheckout-header">
        <div class="checkoutHeader-content">
            <h1 class="is-srOnly">Checkout</h1>
            <h2 class="checkoutHeader-heading">
                <a class="checkoutHeader-link" href="<%= storeUrl %>/">
                    <span class="header-logo-text">Local Testing Environment</span>
                </a>
            </h2>
        </div>
    </header>
    <div class="ng-checkout-container" id="micro-app-ng-checkout">
        <div class="ng-checkout-view" data-ui-view></div>
    </div>
    <script type="text/javascript">
        window.checkoutConfig = {
            checkoutId: '<%= checkoutId %>',
            containerId: 'micro-app-ng-checkout',
        };
    </script>
    <script src="<%= localhostUrl %>/auto-loader.js" type="text/javascript"></script>
    <script type="text/javascript">
        (function() {
            window.checkout.renderCheckout({
                checkoutId: '<%= checkoutId %>',
                containerId: 'micro-app-ng-checkout',
                sentryConfig: null,
            })
        })();
    </script>
    </body>
</html>
