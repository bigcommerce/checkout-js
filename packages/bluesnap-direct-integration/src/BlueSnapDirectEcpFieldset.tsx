import { LanguageService } from '@bigcommerce/checkout-sdk';
import React, { FunctionComponent, memo } from 'react';

import { CheckboxFormField, Fieldset, Legend } from '@bigcommerce/checkout/ui';

import BlueSnapDirectNumberField from './BlueSnapDirectNumberField';
import BlueSnapDirectSelectField from './BlueSnapDirectSelectField';

export interface BlueSnapDirectEcpFieldsetProps {
    language: LanguageService;
    useFloatingLabel?: boolean;
    onPermissionChange: (shopperPermission: boolean) => void;
}

const BlueSnapDirectEcpFieldset: FunctionComponent<BlueSnapDirectEcpFieldsetProps> = ({
    language,
    useFloatingLabel,
    onPermissionChange,
}) => {
    const options = {
        helperLabel: language.translate('payment.bluesnap_direct_account_type_select.label'),
        items: [
            {
                label: language.translate(
                    'payment.bluesnap_direct_account_type_select.option_consumer_checking',
                ),
                value: 'CONSUMER_CHECKING',
            },
            {
                label: language.translate(
                    'payment.bluesnap_direct_account_type_select.option_consumer_savings',
                ),
                value: 'CONSUMER_SAVINGS',
            },
            {
                label: language.translate(
                    'payment.bluesnap_direct_account_type_select.option_corporate_checking',
                ),
                value: 'CORPORATE_CHECKING',
            },
            {
                label: language.translate(
                    'payment.bluesnap_direct_account_type_select.option_corporate_savings',
                ),
                value: 'CORPORATE_SAVINGS',
            },
        ],
    };

    return (
        <Fieldset
            legend={
                <Legend hidden>
                    {language.translate('payment.bluesnap_direct_electronic_check_label')}
                </Legend>
            }
            style={{ paddingBottom: '1rem' }}
        >
            <BlueSnapDirectNumberField
                labelContent={language.translate('payment.bluesnap_direct_account_number.label')}
                maxLength={17}
                name="accountNumber"
                useFloatingLabel={useFloatingLabel}
            />

            <BlueSnapDirectNumberField
                labelContent={language.translate('payment.bluesnap_direct_routing_number.label')}
                maxLength={9}
                name="routingNumber"
                useFloatingLabel={useFloatingLabel}
            />

            <BlueSnapDirectSelectField
                labelContent={language.translate('payment.bluesnap_direct_account_type.label')}
                name="accountType"
                options={options}
                useFloatingLabel={useFloatingLabel}
            />

            <CheckboxFormField
                labelContent={language.translate('payment.bluesnap_direct_permission')}
                name="shopperPermission"
                onChange={onPermissionChange}
            />
        </Fieldset>
    );
};

export default memo(BlueSnapDirectEcpFieldset);
