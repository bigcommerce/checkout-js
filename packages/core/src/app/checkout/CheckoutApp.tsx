import { createCheckoutService, createEmbeddedCheckoutMessenger, createStepTracker, StepTracker } from '@bigcommerce/checkout-sdk';
import { BrowserOptions } from '@sentry/browser';
import React, { Component } from 'react';
import ReactModal from 'react-modal';

import '../../scss/App.scss';
import { createErrorLogger, ErrorBoundary, ErrorLogger } from '../common/error';
import { createEmbeddedCheckoutStylesheet, createEmbeddedCheckoutSupport } from '../embeddedCheckout';
import { getLanguageService, LocaleProvider } from '../locale';

import Checkout from './Checkout';
import CheckoutProvider from './CheckoutProvider';

export interface CheckoutAppProps {
    checkoutId: string;
    containerId: string;
    publicPath?: string;
    sentryConfig?: BrowserOptions;
}

export default class CheckoutApp extends Component<CheckoutAppProps> {
    private checkoutService = createCheckoutService({
        locale: getLanguageService().getLocale(),
        shouldWarnMutation: process.env.NODE_ENV === 'development',
    });
    private embeddedStylesheet = createEmbeddedCheckoutStylesheet();
    private embeddedSupport = createEmbeddedCheckoutSupport(getLanguageService());
    private errorLogger: ErrorLogger;

    constructor(props: Readonly<CheckoutAppProps>) {
        super(props);

        this.errorLogger = createErrorLogger(
            { sentry: props.sentryConfig },
            {
                errorTypes: ['UnrecoverableError'],
                publicPath: props.publicPath,
            }
        );
    }

    componentDidMount(): void {
        const { containerId } = this.props;

        ReactModal.setAppElement(`#${containerId}`);
    }

    render() {
        return (
            <ErrorBoundary logger={ this.errorLogger }>
                <LocaleProvider checkoutService={ this.checkoutService }>
                    <CheckoutProvider checkoutService={ this.checkoutService }>
                        <Checkout
                            { ...this.props }
                            createEmbeddedMessenger={ createEmbeddedCheckoutMessenger }
                            createStepTracker={ this.createStepTracker }
                            embeddedStylesheet={ this.embeddedStylesheet }
                            embeddedSupport={ this.embeddedSupport }
                            errorLogger={ this.errorLogger }
                        />
                    </CheckoutProvider>
                </LocaleProvider>
            </ErrorBoundary>
        );
    }

    private createStepTracker: () => StepTracker = () => {
        return createStepTracker(this.checkoutService);
    };
}
