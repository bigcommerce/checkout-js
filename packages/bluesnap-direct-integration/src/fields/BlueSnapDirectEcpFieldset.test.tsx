import { createLanguageService } from '@bigcommerce/checkout-sdk';
import { Formik } from 'formik';
import { noop } from 'lodash';
import React from 'react';

import { render } from '@bigcommerce/checkout/test-utils';

import BlueSnapDirectEcpFieldset, {
    BlueSnapDirectEcpFieldsetProps,
} from './BlueSnapDirectEcpFieldset';

describe('BlueSnapDirectEcpFieldset', () => {
    let initialValues: { [key: string]: unknown };
    let options: BlueSnapDirectEcpFieldsetProps;

    beforeEach(() => {
        initialValues = {
            accountNumber: '',
            routingNumber: '',
        };
        options = {
            language: createLanguageService(),
        };
    });

    it('matches snapshot', () => {
        expect(
            render(
                <Formik initialValues={initialValues} onSubmit={noop}>
                    <BlueSnapDirectEcpFieldset {...options} />
                </Formik>,
            ),
        ).toMatchSnapshot();
    });
});
