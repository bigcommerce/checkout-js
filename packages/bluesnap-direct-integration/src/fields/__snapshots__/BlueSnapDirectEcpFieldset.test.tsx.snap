// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlueSnapDirectEcpFieldset matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <fieldset
        class="form-fieldset"
        style="padding-bottom: 1rem;"
      >
        <legend
          class="form-legend is-srOnly"
        >
          optimized_checkout.payment.bluesnap_direct_electronic_check_label
        </legend>
        <div
          class="form-body"
        >
          <div
            class="form-field"
          >
            <label
              class="form-label optimizedCheckout-form-label"
              for="accountNumber"
              id="accountNumber-label"
            >
              optimized_checkout.payment.bluesnap_direct_account_number.label
            </label>
            <input
              aria-labelledby="accountNumber-label accountNumber-field-error-message"
              autocomplete="cc-number"
              class="form-input optimizedCheckout-form-input"
              data-test="accountNumber-text"
              id="accountNumber"
              maxlength="17"
              name="accountNumber"
              type="tel"
              value=""
            />
          </div>
          <div
            class="form-field"
          >
            <label
              class="form-label optimizedCheckout-form-label"
              for="routingNumber"
              id="routingNumber-label"
            >
              optimized_checkout.payment.bluesnap_direct_routing_number.label
            </label>
            <input
              aria-labelledby="routingNumber-label routingNumber-field-error-message"
              autocomplete="cc-number"
              class="form-input optimizedCheckout-form-input"
              data-test="routingNumber-text"
              id="routingNumber"
              maxlength="9"
              name="routingNumber"
              type="tel"
              value=""
            />
          </div>
          <div
            class="form-field"
          >
            <label
              class="form-label optimizedCheckout-form-label"
              for="accountType"
              id="accountType-label"
            >
              optimized_checkout.payment.bluesnap_direct_account_type.label
            </label>
            <div
              class="dropdown-chevron"
            >
              <div
                class="icon"
              >
                <svg
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </div>
            </div>
            <select
              aria-labelledby="accountType-label accountType-field-error-message"
              class="form-select optimizedCheckout-form-select"
              data-test="accountType-select"
              id="accountType"
              name="accountType"
            >
              <option
                value=""
              >
                optimized_checkout.payment.bluesnap_direct_account_type_select.label
              </option>
              <option
                value="CONSUMER_CHECKING"
              >
                optimized_checkout.payment.bluesnap_direct_account_type_select.option_consumer_checking
              </option>
              <option
                value="CONSUMER_SAVINGS"
              >
                optimized_checkout.payment.bluesnap_direct_account_type_select.option_consumer_savings
              </option>
              <option
                value="CORPORATE_CHECKING"
              >
                optimized_checkout.payment.bluesnap_direct_account_type_select.option_corporate_checking
              </option>
              <option
                value="CORPORATE_SAVINGS"
              >
                optimized_checkout.payment.bluesnap_direct_account_type_select.option_corporate_savings
              </option>
            </select>
          </div>
          <div
            class="form-field"
          >
            <input
              class="form-checkbox optimizedCheckout-form-checkbox"
              id="shopperPermission"
              name="shopperPermission"
              type="checkbox"
              value=""
            />
            <label
              class="form-label optimizedCheckout-form-label"
              for="shopperPermission"
            >
              optimized_checkout.payment.bluesnap_direct_permission
            </label>
          </div>
        </div>
      </fieldset>
    </div>
  </body>,
  "container": <div>
    <fieldset
      class="form-fieldset"
      style="padding-bottom: 1rem;"
    >
      <legend
        class="form-legend is-srOnly"
      >
        optimized_checkout.payment.bluesnap_direct_electronic_check_label
      </legend>
      <div
        class="form-body"
      >
        <div
          class="form-field"
        >
          <label
            class="form-label optimizedCheckout-form-label"
            for="accountNumber"
            id="accountNumber-label"
          >
            optimized_checkout.payment.bluesnap_direct_account_number.label
          </label>
          <input
            aria-labelledby="accountNumber-label accountNumber-field-error-message"
            autocomplete="cc-number"
            class="form-input optimizedCheckout-form-input"
            data-test="accountNumber-text"
            id="accountNumber"
            maxlength="17"
            name="accountNumber"
            type="tel"
            value=""
          />
        </div>
        <div
          class="form-field"
        >
          <label
            class="form-label optimizedCheckout-form-label"
            for="routingNumber"
            id="routingNumber-label"
          >
            optimized_checkout.payment.bluesnap_direct_routing_number.label
          </label>
          <input
            aria-labelledby="routingNumber-label routingNumber-field-error-message"
            autocomplete="cc-number"
            class="form-input optimizedCheckout-form-input"
            data-test="routingNumber-text"
            id="routingNumber"
            maxlength="9"
            name="routingNumber"
            type="tel"
            value=""
          />
        </div>
        <div
          class="form-field"
        >
          <label
            class="form-label optimizedCheckout-form-label"
            for="accountType"
            id="accountType-label"
          >
            optimized_checkout.payment.bluesnap_direct_account_type.label
          </label>
          <div
            class="dropdown-chevron"
          >
            <div
              class="icon"
            >
              <svg
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </div>
          </div>
          <select
            aria-labelledby="accountType-label accountType-field-error-message"
            class="form-select optimizedCheckout-form-select"
            data-test="accountType-select"
            id="accountType"
            name="accountType"
          >
            <option
              value=""
            >
              optimized_checkout.payment.bluesnap_direct_account_type_select.label
            </option>
            <option
              value="CONSUMER_CHECKING"
            >
              optimized_checkout.payment.bluesnap_direct_account_type_select.option_consumer_checking
            </option>
            <option
              value="CONSUMER_SAVINGS"
            >
              optimized_checkout.payment.bluesnap_direct_account_type_select.option_consumer_savings
            </option>
            <option
              value="CORPORATE_CHECKING"
            >
              optimized_checkout.payment.bluesnap_direct_account_type_select.option_corporate_checking
            </option>
            <option
              value="CORPORATE_SAVINGS"
            >
              optimized_checkout.payment.bluesnap_direct_account_type_select.option_corporate_savings
            </option>
          </select>
        </div>
        <div
          class="form-field"
        >
          <input
            class="form-checkbox optimizedCheckout-form-checkbox"
            id="shopperPermission"
            name="shopperPermission"
            type="checkbox"
            value=""
          />
          <label
            class="form-label optimizedCheckout-form-label"
            for="shopperPermission"
          >
            optimized_checkout.payment.bluesnap_direct_permission
          </label>
        </div>
      </div>
    </fieldset>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
